packages/jokul/src/components-beta/search/Search.tsx

```diff
@@ -0,0 +1,106 @@
+import clsx from "clsx";
+import React, { forwardRef, useRef } from "react";
+import { InputGroup } from "../../components/input-group/index.js";
+import { SearchButton } from "./SearchButton.js";
+import type { SearchInputProps } from "./types.js";
+
+export const Search = forwardRef<HTMLInputElement, SearchInputProps>(
+    function Search(props, forwardedRef): React.JSX.Element {
+        const {
+            label = "Søk",
+            className,
+            density,
+            errorLabel,
+            helpLabel,
+            labelProps = { srOnly: true },
+            supportLabelProps,
+            tooltip,
+            description,
+            name = "q",
+            placeholder = "Søk",
+            showClearButton = true,
+            required = true,
+            spellCheck = false,
+            icon = "search",
+            children,
+            value,
+            ...rest
+        } = props;
+
+        const inputGroupProps = {
+            label,
+            density,
+            errorLabel,
+            helpLabel,
+            labelProps,
+            supportLabelProps,
+            tooltip,
+            description,
+        };
+
+        const internalRef = useRef<HTMLInputElement | null>(null);
+
+        const ref = (instance: HTMLInputElement | null) => {
+            internalRef.current = instance;
+            if (forwardedRef) {
+                if (typeof forwardedRef === "function") {
+                    forwardedRef(instance);
+                } else {
+                    forwardedRef.current = instance;
+                }
+            }
+        };
+
+        const handleClick = () => {
+            if (internalRef.current) {
+                const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+                    window.HTMLInputElement.prototype,
+                    "value",
+                )?.set;
+                nativeInputValueSetter?.call(internalRef.current, "");
+                const event = new Event("change", { bubbles: true });
+                internalRef.current.dispatchEvent(event);
```

Comment:
Lurer på om vi bør sette fokus tilbake på feltet når det blir tømt? Tenkte ikke over det mens vi jobbet med å få til tømming, men merker at jeg forventer det når jeg forsøker å bruke komponenten, særlig med tastaturnavigasjon.


```suggestion
                internalRef.current.dispatchEvent(event);
                internalRef.current.focus();
```